docs/source/configuration/authn-jwt.mdx

```diff
@@ -3,40 +3,62 @@ title: JWT Authentication in the Apollo Router
-
-```yaml title="router.yaml"
-authentication:
-  experimental:
-    jwt:
-      jwks_urls:
-        - https://dev-zzp5enui.us.auth0.com/.well-known/jwks.json
-      # These are the DEFAULT values for the following keys:
-      header_name: Authorization
-      header_value_prefix: Bearer
-      cooldown: 15s
-```
+1. Whenever a client authenticates with your system, your IdP issues that client a valid JSON Web Token (JWT).
+2. In its subsequent requests to your router, the authenticated client provides its JWT in a designated HTTP header.
+3. Whenever your router receives a client request, it extracts the JWT from the designated header (if present).
+4. Your router validates the extracted JWT using a corresponding [JSON Web Key](https://www.rfc-editor.org/rfc/rfc7517) (**JWK**).
+    - Your router obtains all of its known JWKs from URLs that you specify in its configuration file. Each URL provides its keys within a single JSON object called a [JWK Set](https://www.rfc-editor.org/rfc/rfc7517#section-5) (or a **JWKS**).
+    - Validation fails if the JWT is malformed, or if it's been expired for more than 60 seconds (this window accounts for synchronization issues).
+
+5. The router extracts all **claims** from the validated JWT and includes them in the request's context, making them available to your [router customizations](../customizations/overview/), such as Rhai scripts.
+6. Your customizations can handle the request differently depending on the details of the extracted claims, and/or you can propagate the claims to subgraphs to enable more granular access control.
+    - For examples, [see below](#working-with-jwt-claims).
+
+## Turning it on
+
+You enable JWT authentication for your router with the following steps:
+
+1. Set configuration options for JWT authentication in your router's [YAML config file](./overview/#yaml-config-file), under the `authentication.experimental` key:
+
+    ```yaml title="router.yaml"
+    authentication:
+      experimental:
+        jwt:
+          jwks_urls: # This key is required.
+            - https://dev-zzp5enui.us.auth0.com/.well-known/jwks.json
+
+          # These keys are optional. Default values are shown.
+          header_name: Authorization
+          header_value_prefix: Bearer
+          cooldown: 15s
+    ```
+
+    These options are documented [below](#configuration-options).
+
+2. Pass all of the following to the `router` executable on startup:
+
+    - The path to the router's YAML configuration file (via the `--config` option)
+    - The graph ref for the [GraphOS variant](/graphos/graphs/overview#variants) your router should use (via the `APOLLO_GRAPH_REF` environment variable)
+    - A [graph API key](/graphos/api-keys/graph-api-keys) that enables the router to authenticate with GraphOS to fetch its supergraph schema (via the `APOLLO_KEY` environment variable)
+
```

Comment:
I don't if this is worth documenting, but mentioning anyway...

You don't need an APOLLO_KEY to enable JWT authentication, just APOLLO_GRAPH_REF. If you don't have a valid graph to use for testing or evaluation, it doesn't even have to be set to a valid GRAPH variant.

So, really all that's required is:
```
APOLLO_GRAPH_REF= ./router <rest of your router command>
```
NOTE: In the illustration above, I'm setting APOLLO_GRAPH_REF in my environment with an empty value. That's the absolute minimum requirement.

If you do set APOLLO_GRAPH_REF and APOLLO_KEY, then the router will start reporting information to Studio and that may not be desirable for evaluation scenarios.

Reply:
Cool, yeah we'll probably end up documenting this nuance with the final docs release, but I reckon it's fine leaving it as-is for the time being

Reply:
Agree it's fine how it is for now, and I'm pretty sure it's best to keep it way even after the release because _generally speaking_ ,`APOLLO_GRAPH_REF` isn't set without `APOLLO_KEY` in our platform.  I think there's value in suggesting the setting of both for consistency across the platform, and we'll need both in the future.